cli/determined_cli/tunnel.py

```diff
@@ -32,24 +71,24 @@ def run(self) -> None:
         try:
             while True:
                 try:
-                    buf = os.read(self.src, 4096)
+                    buf = self.src.read(4096)
                 except OSError:
```

Comment:
Did you confirm that this is the correct error to check for, even with the wrapped tls connections?  I would expect them to puke with some other error.

Reply:
Ah, thanks for reminding me. A read or write on a closed socket is still `OSError` for the wrapped sockets. Are there any other situations you can think of that I should check for?

Reply:
I'd make sure that if you `docker kill` the container or `kill` the determined master a few times each that you don't get any weird messages on shutdown.  I've definitely seen ssl errors when I closed ssl-wrapped sockets while a thread was actively trying to `read` from them in an e2e test for a personal project, but it isn't every time.

Reply:
I tried killing some things and didn't see anything popping out. I also just noticed that all of the exceptions that the `ssl` module documents are [subclasses of `OSError`](https://docs.python.org/3/library/ssl.html#exceptions), so everything should be covered.